generator.py

```diff
@@ -46,8 +50,25 @@ def sick_days_filter(day):
     logger.info("A user entered: " + user_choice)
 
     if user_choice.lower() == 'register':
-        register_user()
-        break
+        new_user = input("Please input your desired user name:\n")
+        advance = UserDataService().check_if_user_unique(new_user)
```

Comment:
I'd rename this variable to `usernameExists` (check my comment to rename the method of UserDataService) to let the reader know why we're "advancing" into the if-body. After renaming the variable it will read:
```
if not usernameExists:
```

Reply:
Done, and checked via running the code with both a new and old username. Got expected results.